From a52a294045cb05acb141d1acfe2a8ef38e611faa Mon Sep 17 00:00:00 2001
From: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
Date: Tue, 29 Nov 2022 16:48:59 +0530
Subject: [PATCH] Enable Control Flow Integrity for Clang 12 and above

Control Flow Integrity(CFI) is mandatory kernel integrity config and its
not enabled.

Fix the issue by enabling CFI for Clang 12 and above.

Change-Id: Ibcf4e6e7afb7eb59a09b9b7ad6a3112d875ab712
Tracked-On: OAM-105089
Signed-off-by: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
---
 arch/x86/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/x86/Kconfig b/arch/x86/Kconfig
index 56a0370d90e3..541feb92d555 100644
--- a/arch/x86/Kconfig
+++ b/arch/x86/Kconfig
@@ -109,7 +109,7 @@ config X86
 	select ARCH_SUPPORTS_KMAP_LOCAL_FORCE_MAP	if NR_CPUS <= 4096
 	select ARCH_SUPPORTS_LTO_CLANG
 	select ARCH_SUPPORTS_LTO_CLANG_THIN
-	select ARCH_SUPPORTS_CFI_CLANG		if X86_64 && CLANG_VERSION >= 130000
+	select ARCH_SUPPORTS_CFI_CLANG		if X86_64 && CLANG_VERSION >= 120000
 	select ARCH_USE_BUILTIN_BSWAP
 	select ARCH_USE_MEMTEST
 	select ARCH_USE_QUEUED_RWLOCKS
-- 
2.38.1

